softlayer/resource_softlayer_dns_domain_record.go

```diff
@@ -178,30 +274,55 @@ func resourceSoftLayerDnsDomainRecordRead(d *schema.ResourceData, meta interface
 	}
 
 	d.Set("data", *result.Data)
-	d.Set("domainId", *result.DomainId)
-	d.Set("expire", *result.Expire)
+	d.Set("domain_id", *result.DomainId)
 	d.Set("host", *result.Host)
-	d.Set("id", *result.Id)
-	d.Set("minimum", *result.Minimum)
-	d.Set("mxPriority", *result.MxPriority)
-	d.Set("refresh", *result.Refresh)
-	d.Set("responsiblePerson", *result.ResponsiblePerson)
-	d.Set("retry", *result.Retry)
-	d.Set("ttl", *result.Ttl)
 	d.Set("type", *result.Type)
+	d.Set("ttl", *result.Ttl)
+
+	if result.Expire != nil {
+		d.Set("expire", *result.Expire)
+	}
+
+	if result.Minimum != nil {
+		d.Set("minimum_ttl", *result.Minimum)
+	}
+
+	if result.MxPriority != nil {
+		d.Set("mx_priority", *result.MxPriority)
+	}
+
+	if result.Refresh != nil {
+		d.Set("refresh", *result.Refresh)
+	}
+
+	if result.ResponsiblePerson != nil {
+		d.Set("responsible_person", *result.ResponsiblePerson)
+	}
+
+	if result.Retry != nil {
+		d.Set("retry", *result.Retry)
+	}
```

Comment:
Do we need to consider that any of the above fields could have changed from non-nil to nil, in which case we should update the resource to reflect the new `nil` value?